core/src/datayoga_core/blocks/relational/write/block.py

```diff
@@ -59,6 +59,25 @@ def init(self, context: Optional[Context] = None):
 
     async def run(self, data: List[Dict[str, Any]]) -> BlockResult:
         logger.debug(f"Running {self.get_block_name()}")
+
+        try:
+            # try to process all records together
+            return await self._run(data)
+        except Exception as e:
+            if len(data) == 1:
```

Comment:
why == 1? what if we have more records?

Reply:
If we have only one record in batch there is no need to retry it, we could throw exception immediately.